game/script-introduction.rpy

```diff
@@ -0,0 +1,396 @@
+init 0 python in jn_introduction:
+    from Enum import Enum
+    import random
+    import store
+    import store.jn_utils
+
+    class JNIntroductionStates(Enum):
+        """
+        Different introduction sequences states/phases; we use these to track progress
+        """
+        new_game = 1
+        first_meeting = 2
+        collecting_thoughts = 3
+        calmed_down = 4
+        acceptance = 5
+        complete = 6
+
+        def __int__(self):
+            return self.value
+
+    INTRODUCTION_STATE_LABEL_MAP = {
+        JNIntroductionStates.new_game: "introduction_opening",
+        JNIntroductionStates.first_meeting: "introduction_first_meeting",
+        JNIntroductionStates.collecting_thoughts: "introduction_collecting_thoughts",
+        JNIntroductionStates.calmed_down: "introduction_calmed_down",
+        JNIntroductionStates.acceptance: "introduction_acceptance",
+        JNIntroductionStates.complete: "introduction_exit"
+    }
+
+default persistent.jn_introduction_state = 1
+
+label introduction_progress_check:
+    # Handling for if player decides to quit during the introduction sequence so we don't skip unseen segments
+    if not jn_introduction.JNIntroductionStates(persistent.jn_introduction_state) == jn_introduction.JNIntroductionStates.new_game:
+        play audio static
+        show glitch_garbled_a zorder 99 with vpunch
+        hide glitch_garbled_a
+        $ main_background.appear()
+        $ jn_atmosphere.show_sky(jn_atmosphere.WEATHER_GLITCH, with_transition=False)
+        play music audio.space_classroom_bgm fadein 1
+
+    $ renpy.jump(jn_introduction.INTRODUCTION_STATE_LABEL_MAP.get(jn_introduction.JNIntroductionStates(persistent.jn_introduction_state)))
+
+label introduction_opening:
+    $ config.allow_skipping = False
+    scene black
+    $ renpy.pause(5)
+
+    # Restore attempt #1..
+    $ renpy.display_menu(items=[ ("Restore natsuki.chr", True)], screen="choice_centred_mute")
```

Comment:
Ah, so the reason I went with this approach is because as far as I know, this is the only way to specify a menu style other than the default - which we go with here since an offset menu doesn't make much sense when Natsuki isn't actually onscreen.

I can add a comment to make clear this is a non-standard approach, unless you knew of a way to use Ren'Py menu syntax with a non-standard menu?